### ISSUE TITLE
Form validation with data-abide

### ISSUE THREAD

**Issue by anothergituser** [zurb/foundation-sites#7357]
Same as #7127
Impossible to submit for after validation is ok.
Please check..

**Comment by anothergituser** [comment]
Well, to answer my own problem, edit foundation.abide.js
These are the changes

```javascript
Abide.prototype._events = function() {
    var self = this;
    this.$element
        .off('.abide')
        .on('reset.fndtn.abide', function(e) {
            self.resetForm($(this));
        })
        .on('submit.fndtn.abide', function(e) {
            // e.preventDefault();
            // comment e.preventDefault and returnt the result of self.validateForm
            // just like abide from foundation 5 does
            return self.validateForm(self.$element);
        })
        .find('input, textarea, select')
        .off('.abide')
        .on('blur.fndtn.abide change.fndtn.abide', function(e) {
            if (self.options.validateOn === 'fieldChange') {
                self.validateInput($(e.target), self.$element);
            }
            // self.validateForm(self.$element);
        })
        .on('keydown.fndtn.abide', function(e) {
            // if (settings.live_validate === true && e.which != 9) {
            //   clearTimeout(self.timer);
            //   self.timer = setTimeout(function () {
            //     self.validate([this], e);
            //   }.bind(this), settings.timeout);
            // }
            // self.validateForm(self.$element);
        });

},
```

and the validation function
```javascript
Abide.prototype.validateForm = function($form) {
    var self = this,
        inputs = $form.find('input'),
        inputCount = $form.find('input').length,
        counter = 0;

    while (counter < inputCount) {
        self.validateInput($(inputs[counter]), $form);
        counter++;
    }

    // what are all the things that can go wrong with a form?
    if ($form.find('.form-error.is-visible').length || $form.find('.is-invalid-label').length) {
        $form.find('[data-abide-error]').css('display', 'block');
        return false; // don't submit when there are errors
    } else {
        $form.find('[data-abide-error]').css('display', 'none');
        return true; // ok to submit
    }
};
```

Also this is my gulp file in case anyone needs it to compile this
```javascript
var gulp = require('gulp');
var $    = require('gulp-load-plugins')();

var uglify = require('gulp-uglify');
var rename = require('gulp-rename');
var concat = require('gulp-concat');

var sassPaths = [
  'bower_components/foundation-sites/scss',
  'bower_components/motion-ui/src'
];

gulp.task('sass', function() {
  return gulp.src('scss/app.scss')
    .pipe($.sass({
      includePaths: sassPaths
    })
      .on('error', $.sass.logError))
    .pipe($.autoprefixer({
      browsers: ['last 2 versions', 'ie >= 9']
    }))
    .pipe(gulp.dest('../public/css'));
});

gulp.task('js', function(done) {
    gulp.src([
        // './bower_components/foundation-sites/js/**/*.js',
        
        './bower_components/foundation-sites/js/foundation.core.js',
        './bower_components/foundation-sites/js/foundation.abide.js',
        './bower_components/foundation-sites/js/foundation.accordion.js',
        './bower_components/foundation-sites/js/foundation.accordionMenu.js',
        './bower_components/foundation-sites/js/foundation.drilldown.js',
        './bower_components/foundation-sites/js/foundation.dropdown.js',
        './bower_components/foundation-sites/js/foundation.dropdownMenu.js',
        './bower_components/foundation-sites/js/foundation.equalizer.js',
        './bower_components/foundation-sites/js/foundation.interchange.js',
        './bower_components/foundation-sites/js/foundation.magellan.js',
        './bower_components/foundation-sites/js/foundation.offcanvas.js',
        './bower_components/foundation-sites/js/foundation.orbit.js',
        './bower_components/foundation-sites/js/foundation.responsiveMenu.js',
        './bower_components/foundation-sites/js/foundation.responsiveToggle.js',
        './bower_components/foundation-sites/js/foundation.reveal.js',
        './bower_components/foundation-sites/js/foundation.slider.js',
        './bower_components/foundation-sites/js/foundation.sticky.js',
        './bower_components/foundation-sites/js/foundation.tabs.js',
        './bower_components/foundation-sites/js/foundation.toggler.js',
        './bower_components/foundation-sites/js/foundation.tooltip.js',
        './bower_components/foundation-sites/js/foundation.util.box.js',
        './bower_components/foundation-sites/js/foundation.util.keyboard.js',
        './bower_components/foundation-sites/js/foundation.util.mediaQuery.js',
        './bower_components/foundation-sites/js/foundation.util.motion.js',
        './bower_components/foundation-sites/js/foundation.util.nest.js',
        './bower_components/foundation-sites/js/foundation.util.timerAndImageLoader.js',
        './bower_components/foundation-sites/js/foundation.util.touch.js',
        './bower_components/foundation-sites/js/foundation.util.triggers.js',
        
        './bower_components/motion-ui/motion-ui.js',
        './bower_components/what-input/what-input.js',
    ])
    .pipe(concat('foundation.js'))
    // .pipe(uglify())
    .pipe(rename({ extname: '.min.js' }))
    .pipe(gulp.dest('../public/js/'))
    .on('end', done);
});

gulp.task('default', ['sass'], function() {
  gulp.watch(['scss/**/*.scss'], ['sass']);
});
```

Just run ```gulp js``` after making those changes. Also note that 
'./bower_components/foundation-sites/js/**/*.js', will not work since 
 './bower_components/foundation-sites/js/foundation.core.js', needs to be first.

Props for a great framework, i really like you can comment out any component in app.scss and plug in your own and everything still works. Like buttons for example.

**Comment by zurbchris** [comment]
Should be fixed with #7590, please comment and submit pr's to the `abide` branch until 6.1 release.
